frontend/src/components/pages/CampCreation/AddSessionsForm.tsx

```diff
@@ -0,0 +1,172 @@
+import {
+  Box,
+  Button,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  HStack,
+  Input,
+  Text,
+  VStack,
+} from "@chakra-ui/react";
+import React, { Dispatch, SetStateAction } from "react";
+import { CreateCampSession } from "../../../types/CampsTypes";
+import SessionDayButton from "./SessionDayButton";
+
+type AddSessionsFormProps = {
+  scheduledSessions: CreateCampSession[];
+  setScheduledSessions: Dispatch<SetStateAction<CreateCampSession[]>>;
+  setShowAddSessions: Dispatch<SetStateAction<boolean>>;
+};
+
+const AddSessionsForm = ({
+  scheduledSessions,
+  setScheduledSessions,
+  setShowAddSessions,
+}: AddSessionsFormProps): JSX.Element => {
+  const [startDate, setStartDate] = React.useState<Date>(new Date());
+  const [successiveSessions, setSuccessiveSessions] = React.useState<number>(0);
+
+  const emptyWeekDays = new Map<string, boolean>([
+    ["Su", false],
+    ["Mo", false],
+    ["Tu", false],
+    ["We", false],
+    ["Th", false],
+    ["Fr", false],
+    ["Sa", false],
+  ]);
+  const [weekDays, setWeekDays] = React.useState<Map<string, boolean>>(
+    new Map<string, boolean>(emptyWeekDays),
+  );
+
+  const updateSelectedSessionDays = (day: string) => {
+    const daySelected: boolean = weekDays.get(day) ?? false;
+    const updatedMap = new Map<string, boolean>(
+      weekDays.set(day, !daySelected),
+    );
+    setWeekDays(updatedMap);
+  };
+
+  const noSessionDaysSelected = (): boolean => {
+    let hasNoSelectedDay = true;
+    Array.from(weekDays.values()).forEach((val) => {
```

Comment:
because .values() returns an iterator and .forEach() doesn't exist on iterators https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/values